### PR TITLE
Connect using AD

### PR DESCRIPTION
**Pull request by StickNitro**
Pull request for #453 - primarily for the SqlServerDriver

added domain property to DriverOptions
updated DriverUtils to parse domain from url (i.e. domain/username:password for domain based logins
updated SqlServerDriver to include domain when connecting

### CODE REVIEW

File: src/driver/DriverOptions.ts [hunk 1]
```diff
@@ -79,4 +79,8 @@ export interface DriverOptions {
      */
     readonly tablesPrefix?: string;
 
+    /**
+     * Specify the domain for active directory connections
```
Comment (pleerock) [inline src/driver/DriverOptions.ts]: Please add dot at the end of comment. Also please add note that this option is mssql-specific.

**Comment by pleerock** [comment]
Thank you for the PR! Please check why tests are failing. Also please create one another PR against 0.0.x-version branch

**Comment by StickNitro** [comment]
Closing PR as identified that you are able to pass ```"extra":{"domain": "mydomain"}``` which do actually get passed down to the driver

**Comment by pleerock** [comment]
I think you can reopen this PR, because even `"extra":{"domain": "mydomain"}` is supported this will be handy if this option will be on the top level

**Comment by pleerock** [comment]
I still insist you to recreate this PR. This would be very useful if we will have domain right in mssql driver options. Please do it if you have a free time.

**Pull request closed without merging**